app/scripts/lib/util.js

```diff
@@ -36,7 +30,20 @@ const getEnvironmentType = (url = window.location.href) => {
   } else {
     return ENVIRONMENT_TYPE_BACKGROUND
   }
-}
+})
+
+/**
+ * Returns the window type for the application
+ *
+ *  - `popup` refers to the extension opened through the browser app icon (in top right corner in chrome and firefox)
+ *  - `fullscreen` refers to the main browser window
+ *  - `notification` refers to the popup that appears in its own window when taking action outside of metamask
+ *  - `background` refers to the background page
+ *
+ * @param {string} [url] - the URL of the window
+ * @returns {string} the environment ENUM
+ */
+const getEnvironmentType = (url = window.location.href) => getEnvironmentTypeMemo(url)
```

Comment:
Nit: Now that we're memoizing this URL, we're introducing the possibility of this being a memory leak if this was ever called on a non-extension URL. To make sure that doesn't happen, maybe we should remove this parameter altogether? (it's not currently used).

Or at the very least we should specify in this doc comment that this should only be called for internal URLs.